src/galileo/utils/singleton.py

```diff
@@ -76,7 +77,12 @@ def _get_key(
         return key + (project, log_stream)
 
     def get(
-        self, *, project: Optional[str] = None, log_stream: Optional[str] = None, experiment_id: Optional[str] = None
+        self,
+        *,
+        project: Optional[str] = None,
+        log_stream: Optional[str] = None,
+        experiment_id: Optional[str] = None,
+        local_metrics: Optional[list[LocalMetricConfig]] = None,
```

Comment:
so we want to cache logger instance using `local_metrics` as well?

Reply:
yes correct, so they get computed on `flush`

Reply:
oh, no, we don't want the local_metrics in the cache key if that's what you're asking.  They should be set at `init` but not used to access the logger